app/controllers/application_controller.rb

```diff
@@ -321,8 +321,10 @@ def saved_redirect_url_or(default)
   end
 
   def generic_exception(exception)
-    Foreman::Logging.exception("Action failed", exception)
-    render :template => "common/500", :layout => !request.xhr?, :status => :internal_server_error, :locals => { :exception => exception}
+    exception = Foreman::WrappedException.new(exception, exception.message) unless exception.is_a? Foreman::Exception
+    ex_message = exception.message
```

Comment:
This actually copies the message twice, the second parameter for the WrappedException is meant for additional message, for example:

    raise ::Foreman::WrappedException.new exception, N_("Unable to connect to LDAP server")

Honestly, I am not sure why you are wrapping this.